This patch was fed back to the upstream community:
http://code.google.com/p/libarchive/issues/detail?id=287
Given the response "I just committed your patch.", it should
no longer be needed with the next release.

--- libarchive/libarchive/archive_write_set_format_ar.c.orig	Mon Mar 26 19:49:00 2012
+++ libarchive/libarchive/archive_write_set_format_ar.c	Tue Nov 27 17:48:04 2012
@@ -165,7 +165,7 @@
 	 * Reject files with empty name.
 	 */
 	pathname = archive_entry_pathname(entry);
-	if (*pathname == '\0') {
+	if (pathname == NULL || *pathname == '\0') {
 		archive_set_error(&a->archive, EINVAL,
 		    "Invalid filename");
 		return (ARCHIVE_WARN);
